wordpress-mobile/AztecEditor-Android
# Add reordering and filtering of toolbar buttons

### Fix
https://github.com/wordpress-mobile/AztecEditor-Android/issues/955

This PR introduces functionality to reorder and filter toolbar buttons. By default it should work as it works now. In addition to this the `Aztec` class now accepts a configuration object `ToolbarOrder` which defines the order of actions in the bottom bar. In addition to this it supports moving the `plugins` block anywhere between the actions and vertical dividers. 

In the implementation I've extracted all the inner views from the toolbar into separate layouts which are later inflated based on the input. The default behaviour is the same as it is now. The changes included changing all the places which require a certain button to be present. 

When testing make sure the current interface is not broken!


### Test
1. Make sure the toolbar `advanced` mode is set to **false** (that's the default setting)
2. Run the test app
3. Test the bottom toolbar
4. Notice it contains the same actions as before
5. Notice all the actions work as expected



### Test
1. Make sure the toolbar `advanced` mode is set to **true** 
2. Run the test app
3. Test the bottom toolbar
4. Notice it contains the same actions as before
5. Notice all the actions work as expected


### Test
1. Make sure the toolbar `advanced` mode is set to **false** (that's the default setting)
2. Change the `Aztec` invocation by adding a `ToolbarOrder.BasicOrder` parameter
3. Add all the `ToolbarElements` you want in order you expect
4. Add the plugins block in the list
5. Run the test app
6. Notice the bottom bar contains the actions you defined in the order you set
7. Check the actions work as expected


### Test
1. Make sure the toolbar `advanced` mode is set to **false** (that's the default setting)
2. Change the `Aztec` invocation by adding a `ToolbarOrder.BasicOrder` parameter
3. Add all the `ToolbarElements` you want in order you expect
4. Do not add the plugins block in the list
5. Run the test app
6. Notice the bottom bar contains the actions you defined in the order you set
7. Notice the plugins are automatically added at the end of the list


### Test
1. Make sure the toolbar `advanced` mode is set to **true** (that's the default setting)
2. Change the `Aztec` invocation by adding a `ToolbarOrder.AdvancedOrder` parameter
3. Add all the `ToolbarElements` you want in order you expect
4. Notice you can add `collapsedItems` and `expandedItems` - expanded items are always visible, collapsed items are opened on the `expand` click
4. Add the plugins block to the `expandedItems` 
5. Run the test app
6. Notice the bottom bar contains the actions you defined in the order you set
7. Check the actions work as expected


### Test
1. Make sure the toolbar `advanced` mode is set to **true** (that's the default setting)
2. Change the `Aztec` invocation by adding a `ToolbarOrder.AdvancedOrder` parameter
3. Add all the `ToolbarElements` you want in order you expect
4. Notice you can add `collapsedItems` and `expandedItems` - expanded items are always visible, collapsed items are opened on the `expand` click
4. Do not add the plugins block
5. Run the test app
6. Notice the bottom bar contains the actions you defined in the order you set
7. Notice the plugins are automatically added at the end of the expanded list


### Review
@[USER_NAME]

Make sure strings will be translated:
